container/lxd/lxd.go

```diff
@@ -27,6 +27,23 @@ var (
 
 const lxdDefaultProfileName = "default"
 
+// ApplyConstraints takes any supported constraints in cons and sets their
+// respective values in the given lxd metadata map.
+func ApplyConstraints(metadata map[string]string, cons constraints.Value) {
+	// apply any constraints
+	if cons.CpuCores != nil {
+		metadata["limits.cpu"] = fmt.Sprintf("%v", *cons.CpuCores)
+	}
+
+	if cons.CpuPower != nil {
+		metadata["limits.cpu.allowance"] = fmt.Sprintf("%v", *cons.CpuPower)
```

Comment:
I didn't do a full audit of all the limits available; should we also support:

    limits.cpu.priority

and perhaps there are others too.